### PR TITLE
bnb address validator

### PR DESCRIPTION
**Pull request by Rakeally**

### CODE REVIEW

File: src/coin_code.ts [hunk 1]
```diff
@@ -6,6 +6,8 @@ enum CoinCode {
   SOL = 'SOL',
   TRX = 'TRX',
   XLM = 'XLM',
+  BNB = 'BNB',
+  BSC = 'ETH',
```
Comment (baahkusi) [inline src/coin_code.ts:10]: change to `BSC`

File: src/coins.ts [hunk 2]
```diff
@@ -1,4 +1,6 @@
 import Coin from './coin.interface';
+import binanceBeaconChain from './coins/binance_beacon_chain';
```
Comment (baahkusi) [inline src/coins.ts:2]: use `CamelCase` for class names

File: src/coins/binance_beacon_chain.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import Coin from '../coin.interface';
+const WAValidator = require('trezor-address-validator');
+
+export default class binanceBeaconChain implements Coin {
```
Comment (baahkusi) [inline src/coins/binance_beacon_chain.ts:4]: use `CamelCase` for class names

File: src/coins/binance_smart_chain.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import Coin from '../coin.interface';
+const WAValidator = require('trezor-address-validator');
+
+export default class binanceSmartChain implements Coin {
```
Comment (baahkusi) [inline src/coins/binance_smart_chain.ts:4]: use `CamelCase` for class names

**Review by baahkusi — CHANGES_REQUESTED** [review]

File: src/coin_code.ts (hunk 1 above)
Comment (Rakeally) [reply src/coin_code.ts:10]: if I do, the test fails. Cause the address is recognized as an ETH address. The tress dependency and others return the BSC address as ETH, so the test keeps failing if I set `BSC = BSC`

File: src/coins/binance_smart_chain.ts (hunk 4 above)
Comment (Rakeally) [reply src/coins/binance_smart_chain.ts:4]: It is already in camel case, do you mean I should use Pascal case?

File: src/coin_code.ts (hunk 1 above)
Comment (baahkusi) [reply src/coin_code.ts:10]: Figure it out.

File: src/coins/binance_smart_chain.ts (hunk 4 above)
Comment (baahkusi) [reply src/coins/binance_smart_chain.ts:4]: i don't know the difference but it should be `BinanceSmartChain`

**Pull request merged**